keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaTaskMonitorAgent.kt

```diff
@@ -43,36 +45,51 @@ class OrcaTaskMonitorAgent(
 
   // 1. Get active tasks from task tracking table
   // 2. For each task, call orca and ask for details
-  // 3. For each completed task, will emit an event for success/failure
+  // 3. For each completed task, will emit an event for success/failure and delete from the table
   override suspend fun invokeAgent() {
     coroutineScope {
       taskTrackingRepository.getTasks()
         .associate {
           it.subject to
             async {
-              orcaService.getOrchestrationExecution(it.id, DEFAULT_SERVICE_ACCOUNT)
-            }
+              try {
+                orcaService.getOrchestrationExecution(it.id, DEFAULT_SERVICE_ACCOUNT)
+              } catch (e: HttpException) {
+                when (e.isNotFound) {
+                  true -> {
+                    log.warn("Exception ${e.message} has caught while calling orca to fetch status for execution id: ${it.id}" +
+                      " Possible reason: orca is saving info for 2000 tasks/app and this task is older. $e")
```

Comment:
To show the stack trace in the log, the exception needs to be passed explicitly as a final argument, not interpolated into the string:

```suggestion
                      " Possible reason: orca is saving info for 2000 tasks/app and this task is older.", e)
```

Reply:
@lorin thank you for the explanation! I didn't know that :)